flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaConsumerTestBase.java

```diff
@@ -633,7 +654,12 @@ public void runStartFromGroupOffsets() throws Exception {
 		readProps.setProperty("auto.offset.reset", "earliest");
 
 		// the committed group offsets should be used as starting points
-		KafkaTestEnvironment.KafkaOffsetHandler kafkaOffsetHandler = kafkaServer.createOffsetHandler(standardProps);
+		Properties offsetHandlerProps = new Properties();
+		offsetHandlerProps.putAll(standardProps);
+		offsetHandlerProps.setProperty("key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+		offsetHandlerProps.setProperty("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
```

Comment:
I didn't do that because the constructor of `KafkaOffsetHandlerImpl`  has no knowledge of whether the provided `Properties` should be manipulated or not (users of `KafkaOffsetHandlerImpl` provide the properties).

However, I think it would make sense to do what you suggested in the `KafkaOffsetHandlerImpl` if it always just uses `standardProps` instead of a provided properties. In our case that would be completely fine. I'll change it as proposed.